core/src/main/java/org/apache/calcite/rex/RexSimplify.java

```diff
@@ -1993,6 +1993,7 @@ private RexNode simplifyOrs(List<RexNode> terms, RexUnknownAs unknownAs) {
     // or 'X IS NOT NULL' if a simplification takes place (because another 'X<>B' is found)
     final Map<RexNode, RexNode> notEqualsComparisonMap = new HashMap<>();
     final RexLiteral trueLiteral = rexBuilder.makeLiteral(true);
+    boolean removeNull = false;
```

Comment:
I would rather use a flag `hasNullTerm`, and also set it in the `case LITERAL` if the case does not remove it.

Reply:
I remove this logic because I found `simplifyOrs` method will call `flattenOr` eventually which use a `set` to eliminate duplicates. So we don't need to handle multiple `NULL` literal separately.